packages/yoastseo/src/scoring/assessments/readability/SubheadingDistributionTooLongAssessment.js

```diff
@@ -65,25 +68,52 @@ class SubheadingsDistributionTooLong extends Assessment {
 			this._config.countTextIn = __( "characters", "wordpress-seo" );
 		}
 
-		this._subheadingTextsLength = this._subheadingTextsLength.sort( function( a, b ) {
+		this._subheadingTextsLength.subHeadingTexts = this._subheadingTextsLength.subHeadingTexts.sort( function( a, b ) {
 			return b.countLength - a.countLength;
 		} );
 
-		this._tooLongTextsNumber = this.getTooLongSubheadingTexts().length;
-
 		const assessmentResult = new AssessmentResult();
 		assessmentResult.setIdentifier( this.identifier );
 
 		this._hasSubheadings = this.hasSubheadings( paper );
 
+		// Give specific feedback for cases where the post starts with a long text without subheadings.
 		const customCountLength = researcher.getHelper( "customCountLength" );
+		let textPrecedingSubheading1Length = 0;
```

Comment:
maybe something like `lengthOfTextBeforeFirstSubheading` is clearer? I found this name quite hard to decipher 😅

Reply:
We wanted it to be aligned with textPrecedingSubheading1, so that what it it's length is same name but with "length added" :)